pkg/controllers/deprovisioning/drift_test.go

```diff
@@ -173,8 +177,12 @@ var _ = Describe("Drift", func() {
 		ExpectReconcileSucceeded(ctx, deprovisioningController, types.NamespacedName{})
 		wg.Wait()
 
+		// Cascade any deletion of the machine to the node
+		ExpectMachinesCascadeDeletion(ctx, env.Client, machines...)
```

Comment:
Can we generalize ExpectEventuallyDeleted or something?

Reply:
These are good callouts. Will handle in a follow-up PR